server/core/region_option.go

```diff
@@ -55,7 +41,15 @@ func WithPendingPeers(pengdingPeers []*metapb.Peer) RegionCreateOption {
 // WithLearners sets the learners for the region.
 func WithLearners(learners []*metapb.Peer) RegionCreateOption {
 	return func(region *RegionInfo) {
-		region.learners = learners
+		peers := region.meta.GetPeers()
+		for i := range peers {
+			for _, l := range learners {
```

Comment:
Just mention it, why not try to keep the same style of the loop like `for j := range learners`, then use `learners[j]` later instead.

Reply:
only because `l` is shorter than `learners[j]`.